src/transport/CryptoContext.cpp

```diff
@@ -208,26 +223,38 @@ CHIP_ERROR CryptoContext::Decrypt(const uint8_t * input, size_t input_length, ui
     uint8_t AAD[kMaxAADLen];
     uint16_t aadLen = sizeof(AAD);
 
-    VerifyOrReturnError(mKeyAvailable, CHIP_ERROR_INVALID_USE_OF_SESSION_KEY);
     VerifyOrReturnError(input != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrReturnError(input_length > 0, CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrReturnError(output != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
 
     ReturnErrorOnFailure(GetIV(header, IV, sizeof(IV)));
     ReturnErrorOnFailure(GetAdditionalAuthData(header, AAD, aadLen));
 
-    KeyUsage usage = kI2RKey;
+    if (mKeyContext)
```

Comment:
suggest `if (mKeyContext != nullptr)` to make very clear it's an object reference, not a bool

Reply:
Fixed.